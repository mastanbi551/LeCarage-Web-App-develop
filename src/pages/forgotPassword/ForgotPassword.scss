@import "../../styles/variables.scss";

.forgot-password-page {
  &__heading {
    padding: 20px 20px 20px 0;
  }

  &__sub-heading {
    @include FontMontserratAndColorWhite;
    padding: 0 0 20px 0;
    font-weight: 500;
    font-size: 18px;
    text-align: start;
    line-height: 24px;
  }

  &__or-text {
    @include FontMontserratAndColorWhite;
    font-weight: 400;
    font-size: 14px;
    line-height: 17px;
    width: 318px;
  }

  &__container {
    display: flex;
    flex-direction: row;
  }

  &__logo-container {
    flex-basis: 20%;
    padding-top: 3%;
  }
  &__logo-img {
    /*to be updated.*/
    color: transparent;
  }

  &__forgot-password-form {
    padding: 50px;
    margin-left: 10%;
    flex-basis: 50%;

    &__emailid-field {
      padding: 0 20px 20px 0;
    }

    &__mobile-number-field {
      padding: 0 20px 20px 0;
    }

    &__form-fields {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
    }

    &__submit-button {
      margin-top: 6%;
    }
    &__login-redirect {
      width: 318px;
      color: $darkgrey;
      text-align: right;
      padding: 10px;
      @include Body_Copy_Main;
      line-height: 15px;
      align-items: center;
      text-align: right;
      color: $darkgrey;
    }
    .redirect-login {
      color: $primaryWhite;
    }
    .static {
      border: 1px solid $lightgrey;
      background-color: transparent;
      box-sizing: border-box;
      width: 100%;
      height: 40px;
      min-width: 19.875rem;
    }
  }
}
.swal2-popup {
  width: 30%;
}
.swal2-title {
  font-size: 15px;
}
.swal2-confirm {
  padding: 3px 12px;
  font-size: 12px;
}
.swal2-cancel {
  padding: 3px 12px;
  font-size: 12px;
}

.swal2-container.swal2-center > .swal2-popup {
  background-color: $grey;
  color: $primaryWhite;
}

.input-text-field .multiple-icons {
  color: #fff;
}
