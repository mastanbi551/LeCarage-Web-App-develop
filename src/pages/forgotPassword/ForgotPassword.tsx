import React from 'react';
import { Formik } from 'formik';
import { BackgroundImageContainer, InputTextField, PageHeader, LeCarageLogo, CustomButton, Loader } from '../../components';
import { ImagesPath, ConstantStrings } from '../../utils/constants';
import './ForgotPassword.scss';
import { ForgotPasswordSchema } from '../../utils/ValidationSchemas';

interface ForgotPassswordProps {
    handleOnSubmit: (values: {email: string}) => void;
    showLoader: boolean
}

export const ForgotPassword: React.FC<ForgotPassswordProps> = ({
    handleOnSubmit,
    showLoader
}) => {
    return (
        <BackgroundImageContainer imageSource={ImagesPath.bgImageWithCar}>
            <Formik
                initialValues={{ email: '' }}
                validationSchema={ForgotPasswordSchema}
                onSubmit={(values, { setSubmitting }) => {
                    handleOnSubmit(values);
                }}
            >
                {({ handleChange, handleSubmit, values, errors, isValid, handleBlur, touched }) => (
                    <div className='forgot-password-page'>
                        <div className='forgot-password-page__container'>
                            <div className='forgot-password-page__forgot-password-form' >
                                <div className='forgot-password-page__heading'>
                                    <PageHeader text={ConstantStrings.fillInThefield} name='heading' />
                                </div>
                                <h2 className='forgot-password-page__sub-heading'>{ConstantStrings.codeverifyMessage}</h2>
                                <form className='forgot-password-page__forgot-password-form__form-fields' onSubmit={handleSubmit} data-testid="forgot-password-form">
                                    <div className="forgot-password-page__forgot-password-form__emailid-field" data-testid="input-field">
                                        <InputTextField
                                            dataTestId="emailfgt"
                                            name='input-field'
                                            id='email_field'
                                            placeholder={ConstantStrings.enterEmail}
                                            value={values.email}
                                            inputType='text'
                                            requiredOrNot={true}
                                            onChangeText={handleChange('email')}
                                            onBlur={handleBlur('email')}
                                            error={errors.email}
                                            touched={touched.email}
                                        />

                                    </div>
                                    <div className='forgot-password-page__forgot-password-form__submit-button'>
                                        <CustomButton
                                            title="SEND OTP"
                                            dataTestId='button_full'
                                            name={showLoader ? 'cursor-not-allowed button_full': 'button_full'} 
                                            buttonOnClick={handleSubmit}
                                            isDisable={!isValid}
                                        />
                                    </div>
                                    <span className='forgot-password-page__forgot-password-form__login-redirect'>click here to <a href='/login' className='redirect-login'>login </a></span>
                                </form>
                            </div>
                            {
                                showLoader && <Loader />
                            }
                            <div className='forgot-password-page__logo-container'>
                                <LeCarageLogo source={ImagesPath.lecaragelogo} altText="logo" name='logo-page__img' />
                            </div>
                        </div>
                    </div>
                )}
            </Formik>
        </BackgroundImageContainer>
    );

};

