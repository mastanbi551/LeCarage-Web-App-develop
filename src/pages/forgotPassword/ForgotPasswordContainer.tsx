// ForgotPasswordContainer.tsx
import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router';
import { useDispatch } from 'react-redux';
import { forgotPasswordRequest, userEmailId } from '../forgotPassword/forgotPasswordState/ForgotPasswordSlice';
import { ForgotPassword } from './ForgotPassword';
import { useAppSelector } from '../../hooks/Hooks';
import { onLoadingStop } from '../forgotPassword/forgotPasswordState/ForgotPasswordSlice';
import { responseStatus } from '../../utils/constants/ConstantStrings';
import { useScroll } from 'framer-motion';

export const ForgotPasswordContainer = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [showLoader, setShowLoader] = useState(false);

  const { responseData, isLoading } = useAppSelector(
    (state) => state.forgotPasswordReducer
  );

  useEffect(() => {
    if (isLoading) {
      setShowLoader(false);
      if (responseData.status === responseStatus.success && responseData.details.length > 0) {
        dispatch(onLoadingStop());
        localStorage.setItem('uId', responseData.details[0].uId);
        navigate('/otpverification');
      }
    }
  }, [isLoading, responseData]);

  const handleOnSubmit = (values: {email: string}) => {
    setShowLoader(true);
    const params = JSON.stringify({
      email: values.email
    });
    dispatch(userEmailId(values.email));
    dispatch(forgotPasswordRequest(params));
  };

  return <ForgotPassword 
    handleOnSubmit={handleOnSubmit} 
    showLoader={showLoader}
  />;
};
