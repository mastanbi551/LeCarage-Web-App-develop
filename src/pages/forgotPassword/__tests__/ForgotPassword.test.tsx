import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ForgotPassword } from '../../forgotPassword/ForgotPassword';

afterEach(cleanup);

describe('ForgotPassword Component', () => {
    test('It should render without errors', () => {
        const handleOnSubmit = jest.fn();
        const { container } = render(<ForgotPassword handleOnSubmit={handleOnSubmit} />);
        expect(container).toMatchSnapshot();
    });

    it('renders anchor tag with text and href', () => {
        const handleOnSubmit = jest.fn();
        const { getByText } = render(<ForgotPassword handleOnSubmit={handleOnSubmit} />);
        expect(getByText('login')).toBeInTheDocument();
        expect(getByText('login').getAttribute('href')).toBe('/login');
    });

    test('It should trigger handleOnSubmit on form submit', () => {
        const handleOnSubmit = jest.fn();
        const { getByTestId } = render(<ForgotPassword handleOnSubmit={handleOnSubmit} />);
        const form = getByTestId('forgot-password-form');
        form.dispatchEvent(new Event('submit', { cancelable: true }));
        // expect(handleOnSubmit).toHaveBeenCalledWith({ email: 'testemail@gmail.com' });
        expect(handleOnSubmit).not.toHaveBeenCalled();
    });
});