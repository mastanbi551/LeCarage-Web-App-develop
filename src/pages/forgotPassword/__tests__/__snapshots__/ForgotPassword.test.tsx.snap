// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Component It should render without errors 1`] = `
<div>
  <div
    data-testid="background-image-container"
    style="background-image: url(bgimg.jpg); background-size: cover; background-repeat: no-repeat; background-position: center; height: 100vh; overflow: hidden;"
  >
    <div
      class="forgot-password-page"
    >
      <div
        class="forgot-password-page__container"
      >
        <div
          class="forgot-password-page__forgot-password-form"
        >
          <div
            class="forgot-password-page__heading"
          >
            <h1
              class="heading"
            >
              Fill in the below field
            </h1>
          </div>
          <h2
            class="forgot-password-page__sub-heading"
          >
            A verification code will be sent to your email
          </h2>
          <form
            class="forgot-password-page__forgot-password-form__form-fields"
            data-testid="forgot-password-form"
          >
            <div
              class="forgot-password-page__forgot-password-form__emailid-field"
              data-testid="input-field"
            >
              <div
                class="input-text-field"
              >
                <input
                  autocomplete="new-password"
                  autocorrect="off"
                  class="input-field static"
                  data-testid="emailfgt"
                  id="email_field"
                  placeholder="Enter your email address"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="forgot-password-page__forgot-password-form__submit-button"
            >
              <button
                class="button_full"
                data-testid="button_full"
                type="button"
              >
                SEND OTP
              </button>
            </div>
            <span
              class="forgot-password-page__forgot-password-form__login-redirect"
            >
              click here to 
              <a
                class="redirect-login"
                href="/login"
              >
                login 
              </a>
            </span>
          </form>
        </div>
        <div
          class="forgot-password-page__logo-container"
        >
          <img
            alt="logo"
            class="logo-page__img"
            src="lecarageLogo.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
